darktable (master branch) fails to build in MSYS2 if build type is non-Debug

**Describe the bug/issue**

darktable fails to build in MSYS2 if the build type is not Debug (ie, RelWithDebInfo or Release). The error message is as follows:
```
C:/msys64/home/Victor/darktable/src/develop/openmp_maths.h:66:1: error: optimization attribute on 'powf' follows definition but the attribute doesn't match [-Werror=attributes]
   66 | extern float powf(const float x, const float y);
      | ^~~~~~
compilation terminated due to -Wfatal-errors.
```

Since the appearance of the warning depends on the level of optimization, the cause is apparently one of the advanced checks that are triggered only when the compiler attempts to perform an optimization, and which is essentially heuristic and can produce false positives.

We could add something like
`#pragma GCC diagnostic ignored "-Wattributes"`

Or maybe @ralfbrown will find a way not to trigger this warning.

**Which commit introduced the error**

Obviously 0eac582e946f54b97f64726e4b01d677e8fe1349, which added the code that throws the false positive warning.

**Platform**

MSYS2 (GCC 12). There are no problems when building on Linux with GCC 11.

--- a/src/develop/openmp_maths.h
+++ b/src/develop/openmp_maths.h
@@ -60,13 +60,6 @@ extern float logf(const float x);
 
 #endif
 
-#if defined(_OPENMP) && defined(__GNUC__) && __GNUC__ >= 10
-
-#pragma omp declare simd
-extern float powf(const float x, const float y);
-
-#endif
-
 /* Bring our own optimized maths functions because Clang makes dumb shit */
 
 #ifdef _OPENMP

